### DISCUSSION TITLE
how to build an opentelemetry context from tracing current span?

### DISCUSSION THREAD

**Discussion by egtwp** [tokio-rs/tracing-opentelemetry#118] [Q&A] [answered]
Hello,
I would like to propagate my `tracing::Span::current()` in my currier to let my server use it as parent in opentelemetry context propagation.

How can I do? I'm missing something?

That's a very simplified view of what my brain was thinking about:

```rust
#[tracing::instrument]
fn my_fn() {
  use tracing_opentelemetry::OpenTelemetrySpanExt;

  let current_span = tracing:Span::current();
  let current_context = current_span.context(); //But is empty!
  
  //Inject opentelemetry context into currier...
}
```

**Comment by ymgyt** [comment, marked as answer]
Hmm, the code mentioned above seems like it would work well.
For reference, we are actually able to integrate baggage using the following method.

0. [init propagator](https://github.com/ymgyt/syndicationd/blob/cb44f3b8af19eeecf16c500bd4478da28c5576ec/crates/synd_o11y/src/opentelemetry/propagation.rs#L59)
1. [inject request_id in baggage](https://github.com/ymgyt/syndicationd/blob/cb44f3b8af19eeecf16c500bd4478da28c5576ec/crates/synd_term/src/client/mod.rs#L178)
2. [inject with text map propagator](https://github.com/ymgyt/syndicationd/blob/cb44f3b8af19eeecf16c500bd4478da28c5576ec/crates/synd_o11y/src/opentelemetry/propagation.rs#L22)
3. [extract context from request](https://github.com/ymgyt/syndicationd/blob/cb44f3b8af19eeecf16c500bd4478da28c5576ec/crates/synd_api/src/serve/layer/trace.rs#L10)

**Reply by egtwp** [reply]
You right! 
My example code seems to work!

I was misled by the `debug` of `opentelemetry::context` that seems to print an empty object even if is not...

I will check the relative code just to understand more how to debug it.

Thank u!
